core/amber/src/main/python/core/runnables/data_processor.py

```diff
@@ -63,6 +64,7 @@ def process_marker(self, marker: Marker) -> None:
                     self._set_output_state(executor.process_state(marker, port_id))
                 elif isinstance(marker, EndOfInputPort):
                     self._set_output_state(executor.produce_state_on_finish(port_id))
+                    self._switch_context()
```

Comment:
Can you explain in the PR description how this fix is done?

Reply:
Added.